src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -176,14 +176,15 @@ private static void AnalyzeCompilationStart(CompilationStartAnalysisContext cont
             foreach (var type in typeSymbols)
             {
                 // User specific mapping has higher priority than hardcoded one
-                if (userMap.TryGetValue(type.OriginalDefinition, out var suffix) &&
-                    !RoslynString.IsNullOrWhiteSpace(suffix))
+                if (userMap.TryGetValue(type.OriginalDefinition, out var suffix))
                 {
-                    suffixInfo = SuffixInfo.Create(suffix, canSuffixBeCollection: false);
-                    return true;
+                    if (!RoslynString.IsNullOrWhiteSpace(suffix))
+                    {
+                        suffixInfo = SuffixInfo.Create(suffix, canSuffixBeCollection: false);
+                        return true;
+                    }
                 }
-
-                if (hardcodedMap.TryGetValue(type.OriginalDefinition, out suffixInfo))
+                else if (hardcodedMap.TryGetValue(type.OriginalDefinition, out suffixInfo))
```

Comment:
This can be seen as breaking change :confused:

Old behavior was that defining a user map *extends* the hardcoded one.
New behavior is that defining a user map *replaces* the hardcoded one.